assignments/arrays.js

```diff
@@ -58,39 +58,94 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // arr = [1,2,3,4];
 // for (i = 0; i < arr.length; i++) {
-//     arr[i]; // 1,2,3,4
+//	 arr[i]; // 1,2,3,4
 // }
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
+function challenge1(arr){
+	for(let i=0; i<arr.length; i++){
+		if(arr[i].id==33) return arr[i];
+	}
+}
+
+let car = challenge1(inventory);
+
+// or with array methods
+// let car = inventory.filter((elem) => elem.id==33)[0];
+
+console.log(`Car 33 is a ${car.car_year} ${car.car_make} ${car.car_model}` );
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+function challenge2(arr){
+	let res = 0;
+	for(let i=0; i<arr.length; i++){
+		if(arr[i].car_year >= arr[res].car_year) res = i; // using ">=" operator because the last car in the inventory is the youngest one, that has been added to the inventory as the last one among the youngest
+	}
+
+	return arr[res];
+}
+
+// let lastCar = inventory.sort((a, b) => b.car_year-a.car_year)[0];
+let lastCar = challenge2(inventory);
+console.log(lastCar.car_make, lastCar.car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+
+// typical sort:
+// let carModels = inventory.sort((a, b) => (a.car_model > b.car_model)? 1 : (b.car_model > a.car_model) ? -1 : 0);
+let carModels = inventory.sort((a, b) => a.car_model > b.car_model);
```

Comment:
The car models are not listed in alphabetical order, I would try to use an if statement in the sort method to get it to sort correctly

Reply:
Exactly - so I tuned a little bit the one I commented above and fixed the letter-case issue.

Fixed in #2.